crates/typst/src/layout/frame.rs

```diff
@@ -34,6 +34,8 @@ pub struct Frame {
     ///
     /// Determines whether it is a boundary for gradient drawing.
     kind: FrameKind,
+    /// An optional label attached to the frame.
+    label: Option<Label>,
```

Comment:
I want to try and keep frames as lightweight and conceptually simple as possible (the existence of `FrameKind` is actually already unfortunate), so I would prefer for this to be in `GroupItem`. You can generate it via `frame.group(..)`. You can search for references to this function to see how it works.

Reply:
I gave it a go on this approach, I had to make `frame.group(..)` public and also the produced result has more groups than the original ones because of the encapsulation:

using `frame.group`

```
group:
  group: box-label
    group:
group:
  group: block-label
    group:
```

labeling `Frame` directly:

```
group:
  group: box-label
group:
  group: block-label
```

Reply:
Update:

I noticed there aren't more groups, it just shifted up 🤣 :

using `frame.group`

```
group:
  group: box-label
    group:
group:
  group: block-label
    group:
```

labeling `Frame` directly:

```
group:
  group:
    group: box-label
group:
  group:
    group: block-label
```